pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -1539,6 +1573,32 @@ func mergeDashbuilder(service kubernetes.PlatformService, cr *api.KieApp, env ap
 	return env, nil
 }
 
+func overrideKafkaTopicsEnv(cr *api.KieApp, env *api.Environment) {
+	var dataSet []string
+	if cr.Status.Applied.Objects.Servers != nil && cr.Status.Applied.Objects.Servers[0].Kafka != nil {
```

Comment:
checking `cr.Status.Applied.Objects.Servers[0].Kafka` doesn't work here because what if `cr.Status.Applied.Objects.Servers[1].Kafka` is not nil instead? each serverSet should have the ability to set enable Kafka if it wants.

Reply:
Ok